src/main/java/uk/gov/companieshouse/api/interceptor/RolePermissionInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.companieshouse.api.interceptor;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import uk.gov.companieshouse.api.util.security.AuthorisationUtil;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+/**
+ * Checks the request to see if the request contains the required role permission
+ * stored in the header field `ERIC-Authorised-Roles`. Interceptor is only used 
+ * for use when verifying an admin user
```

Comment:
Could the final sentence be reworded to "Interceptor is only for use when verifying an admin user"